core/src/main/scala/chisel3/Printf.scala

```diff
@@ -34,6 +33,9 @@ object printf {
     formatIn map escaped mkString ""
   }
 
+  /** Named class for [[printf]]s. */
+  final class Printf(val pable: Printable) extends BaseSim
+
```

Comment:
@jackkoenig I moved this into the `printf` object, because I was getting really weird errors having it out in the base `chisel3` package:

```
[error] /Users/debs/Documents/chisel3/src/main/scala/chisel3/aop/Select.scala:416:33: type Printf is not a member of package chisel3
[error]   case class Printf(id: chisel3.Printf, preds: Seq[Predicate], pable: Printable, clock: Clock) extends Serializeable {
[error]                                 ^
[error] /Users/debs/Documents/chisel3/src/main/scala/chisel3/compatibility.scala:302:24: object printf is not a member of package chisel3
[error] did you mean Printf?
[error]   val printf = chisel3.printf
[error]       
...
[error] /Users/debs/Documents/chisel3/src/main/scala/chisel3/internal/firrtl/Emitter.scala:81:18: missing argument list for method printf in object Predef
[error] Unapplied methods are only converted to functions when a function type is expected.
[error] You can make this conversion explicit by writing `printf _` or `printf(_,_)` instead of `printf`.
[error]           "\"" + printf.format(fmt) + "\"") ++ args
[error]                  ^
[error] /Users/debs/Documents/chisel3/src/main/scala/chisel3/internal/firrtl/Emitter.scala:84:92: missing argument list for method printf in object Predef
[error] Unapplied methods are only converted to functions when a function type is expected.
[error] You can make this conversion explicit by writing `printf _` or `printf(_,_)` instead of `printf`.
[error]         s"""${e.op}(${e.clock.fullName(ctx)}, ${e.predicate.fullName(ctx)}, UInt<1>(1), "${printf.format(e.message)}") : ${e.name}"""
[error]   
```

Do you think this is a suitable place for this class to live?

Reply:
We could resolve these issues but I kind of like having each of them within their respective objects. It's a bit neater than our usually strategy of dumping everything in package `chisel3`.

Reply:
kk, i'm happy with it if you are! Can you help me merge if everything else looks good?